### DISCUSSION TITLE
Use custom x11 client (ZigX)

### DISCUSSION THREAD

**Discussion by kritma** [godotengine/godot-proposals#8338] [Platforms]
## X11 
* bulky lib
* adds 9 dynamic libs in binary
## ZigX
* better errors
* more control
* not production ready lib 
* statically compiles 
---
[ZIgX presentation](https://youtu.be/aPWFLkHRIAQ)

**Comment by Calinou** [comment]
libX11 is hardly a deterrent for building portable binaries as it stands[^1], so I don't think there's too much of a priority to replace it. Godot is already set in a way that it's easy to build portable binaries if you compile on an old enough Linux distribution (typically the oldest still-supported Ubuntu LTS), or use the https://github.com/godotengine/build-containers.

[^1]: libc and libstdc++ are way larger issues in comparison, but we got the libstdc++ part squared out in Godot's buildsystem. libc is why you need to build on an older Linux distribution or use the build containers though. You could try to use musl, but it can't be used in graphical applications due to NVIDIA libgl stuff and can lead to slower runtime performance.
